### ISSUE TITLE
BUG: Background dataset subsampling

### ISSUE THREAD

**Issue by rozeni89** [shap/shap#3680]
### Issue Description

The background dataset always gets subsampled to 100 samples

### Minimal Reproducible Example

```python
from sklearn.datasets import fetch_openml
import shap
import numpy as np
from sklearn.preprocessing import OrdinalEncoder
from sklearn.compose import ColumnTransformer
from sklearn.model_selection import train_test_split
from sklearn.ensemble import HistGradientBoostingRegressor
from timeit import default_timer as timer

# Load data
df = fetch_openml(data_id=41214, as_frame=True).frame

# Create index
df['IDpol'] = df['IDpol'].astype(np.int64)
df.set_index('IDpol', inplace=True)

# Cap exposure at 1 per original study due to potential data defects
df['Exposure'] = df['Exposure'].clip(upper=1)

# Preporcess data for modeling
cats = ['Area', 'VehBrand', 'VehGas', 'Region', 'VehPower']
nums = ['Exposure','Density', 'VehAge', 'BonusMalus']

ct = ColumnTransformer(
transformers=[
    ('cat', OrdinalEncoder(), cats),
    #('num', StandardScaler(), nums),
],
remainder="passthrough",  verbose_feature_names_out = False
)

ct.set_output(transform="pandas")
df = ct.fit_transform(df)

# Split data into train and test and create target variable
# drop "Density" -- very strongly correlated to "Area"
X = df.drop(['ClaimNb', 'Density'], axis=1)
y = df['ClaimNb']/df['Exposure'] # Create frequency target

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=1
)

offset = X_train.Exposure

X_train.drop('Exposure', axis=1, inplace = True)
X_test.drop('Exposure', axis=1, inplace = True)

# Fit model
model = HistGradientBoostingRegressor(loss="poisson", max_leaf_nodes=128)
model.fit(X_train, y_train, sample_weight=offset)

# Calculate SVs for three different numbers of background samples
v = X_test.iloc[0:1000]

for jj in [100, 1000, 10000]:
    X_sub = shap.sample(X_train, jj)
    ex = shap.TreeExplainer(model, data=X_sub, feature_perturbation='interventional')
   
    start = timer()
    shap_values = ex(v)
    end = timer()
    print(end - start)
```

### Traceback

```shell
No error is produced
```

### Expected Behavior

The times produced by the above code should be significantly different as computational burden increases with number of samples. Yet they come out to be roughly the same. Going through debugging it seems that background set is always downsampled to 100 in "_tabular.py".

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [ ] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

0.42.1

**Comment by CloseChoice** [comment]
Thanks for reporting. Please use

```python
from shap import maskers
...
ex = shap.TreeExplainer(model,
                        data=maskers.Independent(X_sub, max_samples=1000),
                        feature_perturbation='interventional')
```
as a workaround.

**Comment by rozeni89** [comment]
This worked. What's confusing is using this command with "data" argument instead of "masker" one which is what I was trying to unsuccessfully do. Out of curiosity, what is the difference between "shap.sample" and "shap.masker.Indpenendent"? It seems like they are trying to do the same thing i.e. subsampling a dataset.

**Comment by CloseChoice** [comment]
`shap.sample` is just an `sklearn.utils.shuffle` if `len(X) > nsamples`. A masker is an object that accepts data + a binary mask and masks the data accordingly.

**Comment by CloseChoice** [comment]
Can we close this issue @rozeni89
